builders/manage/dapp/services-payment.md

```diff
@@ -0,0 +1,63 @@
+- The account you used when registering the appchain, imported in any of the [supported wallets](/builders/deploy/dapp/#supported-wallets){target=\_blank}
+
+If you're unsure what your registration account is, you can query it directly from the Tanssi orchestrator chain, which keeps records of every registered appchain. To do so, head to the **Chain state** section on the Polkadot.js Apps connected to the orchestrator chain for [snap appchains](https://polkadot.js.org/apps/?rpc=wss://fraa-flashbox-2607-rpc.a.stagenet.tanssi.network#/chainstate){target=\_blank} or [dedicated appchains](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Ffraa-dancebox-rpc.a.dancebox.tanssi.network#/chainstate){target=\_blank}, and take the following steps:
+
+1. Select the **registrar** storage module
+2. Select **registrarDeposit**
+3. Insert you appchain id
+4. Press **+** icon
+
+You'll see the registration account at the bottom.
+
+![Locating your registration account](/images/builders/manage/dapp/services-payment/services-payment-1.webp)
+
+## Topping-Up Your Appchain's Credits {: #topping-up }
+
+Following a pay-as-you-go model, the appchains must have funds allocated to pay for the services, which, in time, will be deducted by the Tanssi protocol upon every session change or every block produced.
+
+!!! note
+    Tanssi reserves 20% of the payments received for the protocol's treasury, and the remainder is burnt.
+
+You can check your appchain's current balance and top it up using the Tanssi dApp. To do so, head to the [Tanssi dApp](https://apps.tanssi.network/){target=\_blank} and connect the appchain's registration account to the dApp. The site will display a card showing your appchain status. This card includes the projected liveness forecast underneath the **Block Production** section and the **Top Up** button.
+
+![Top-up action button in the dApp](/images/builders/manage/dapp/services-payment/services-payment-2.webp)
+
+Clicking on the **Top Up** button displays a sidebar where the following information can be seen:
+
+- **Current balance** - the current balance allocated for the appchain's block production service
+- **Available balance** - the balance available in the appchain's registration account, which is connected to the dApp
+- **Current cost** - current cost per block producer's assignment and cost per block
+- **Projected forecast** - the estimated day when the appchain runs out of funds and stops being served
+
+To extend the appchain's projected runway, take the following actions:
+
+1. Insert the amount of tokens to buy block production services
+2. Top Up
+
+![Top-up sidebar](/images/builders/manage/dapp/services-payment/services-payment-3.webp)
+
+You will be asked to sign the transaction, and after it's gone through, your appchain will benefit from an extended horizon of liveness.
+
+!!! warning
+    If your appchain doesn't have enough funds to cover the block producer's assignment and one session worth of blocks, it will stall.
```

Comment:
```suggestion
    If your appchain doesn't have enough funds to cover the block producer's assignment and one session's worth of blocks, it will stall.
```